typing_extensions/src_py3/typing_extensions.py

```diff
@@ -729,6 +738,30 @@ def _next_in_mro(cls):
                 next_in_mro = cls.__mro__[i + 1]
         return next_in_mro
 
+
+def _get_protocol_attrs(cls):
+    attrs = set()
+    for base in cls.__mro__[:-1]:  # without object
+        if base.__name__ in ('Protocol', 'Generic'):
+            continue
+        annotations = getattr(base, '__annotations__', {})
+        for attr in list(base.__dict__.keys()) + list(annotations.keys()):
+            if (not attr.startswith('_abc_') and attr not in (
+                    '__abstractmethods__', '__annotations__', '__weakref__',
+                    '_is_protocol', '_is_runtime_protocol', '__dict__',
+                    '__args__', '__slots__',
+                    '__next_in_mro__', '__parameters__', '__origin__',
+                    '__orig_bases__', '__extra__', '__tree_hash__',
+                    '__doc__', '__subclasshook__', '__init__', '__new__',
+                    '__module__', '_MutableMapping__marker', '_gorg')):
+                attrs.add(attr)
+    return attrs
+
+
+def _callable_members_only(cls):
```

Comment:
It would be nice if this function followed the (very informal) naming convention for booleans to start with "has" or "is". Initially I mistook it for a filter over `_get_protocol_attrs()`.